sdk/eventhubs/azure-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventDataBatch.java

```diff
@@ -22,52 +23,87 @@
 import java.util.Locale;
 import java.util.Objects;
 
-/*
+/**
  * A class for aggregating EventData into a single, size-limited, batch that will be treated as a single message when
  * sent to the Azure Event Hubs service.
+ *
+ * @see EventHubProducer#createBatch()
+ * @see EventHubProducer#createBatch(BatchOptions)
+ * @see EventHubProducer See EventHubProducer for examples.
  */
-final class EventDataBatch {
+public final class EventDataBatch {
+    private final ClientLogger logger = new ClientLogger(EventDataBatch.class);
+    private final Object lock = new Object();
     private final int maxMessageSize;
     private final String partitionKey;
     private final ErrorContextProvider contextProvider;
     private final List<EventData> events;
     private final byte[] eventBytes;
-    private int currentSize;
+    private int sizeInBytes;
 
     EventDataBatch(int maxMessageSize, String partitionKey, ErrorContextProvider contextProvider) {
         this.maxMessageSize = maxMessageSize;
         this.partitionKey = partitionKey;
         this.contextProvider = contextProvider;
         this.events = new LinkedList<>();
-        this.currentSize = (maxMessageSize / 65536) * 1024; // reserve 1KB for every 64KB
+        this.sizeInBytes = (maxMessageSize / 65536) * 1024; // reserve 1KB for every 64KB
         this.eventBytes = new byte[maxMessageSize];
     }
 
-    int getSize() {
+    /**
+     * Gets the number of {@link EventData events} in the batch.
+     *
+     * @return The number of {@link EventData events} in the batch.
+     */
+    public int getCount() {
         return events.size();
     }
 
-    boolean tryAdd(final EventData eventData) {
+    /**
+     * Gets the size of the {@link EventDataBatch} in bytes.
+     *
+     * @return the size of the {@link EventDataBatch} in bytes.
+     */
+    public int getSizeInBytes() {
+        return this.sizeInBytes;
+    }
 
+    /**
+     * Tries to add an {@link EventData eventData} to the batch.
+     *
+     * @param eventData The {@link EventData} to add to the batch.
+     * @return {@code true} if the event could be added to the batch; {@code false} if the event was too large to fit in
+     *         the batch.
+     * @throws IllegalArgumentException if {@code eventData} is {@code null}.
+     * @throws AmqpException if {@code eventData} is larger than the maximum size of the {@link
+     *         EventDataBatch}.
+     */
+    public boolean tryAdd(final EventData eventData) {
         if (eventData == null) {
-            throw new IllegalArgumentException("eventData cannot be null");
+            logger.logAndThrow(new IllegalArgumentException("eventData cannot be null"));
+            return false;
```

Comment:
If you are throwing error `"eventData cannot be null"`, then `return false` would be unreachable code ?

Reply:
The `logger.logAndThrow` method returns void, and the throw is inside that method, so the compiler doesn't know that it won't reach the next line of code.